force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -71,7 +83,17 @@ public with sharing class FlowControlLoopsCollections {
      * @return A string indicating whether the number is "Positive", "Negative", or "Zero", or null if the number is null.
      */
     public static String checkNumber(Integer a) {
-        return null; // Replace null with the variable you used to store the result
+        String inputSignIndicator;
+        if (a>0) {
+            inputSignIndicator = 'Positive';
+        } else if (a < 0) {
+            inputSignIndicator = 'Negative';
+        } else if (a == 0){
+            inputSignIndicator = 'Zero';
+        } else {
+            inputSignIndicator = null;
```

Comment:
This else section would never be reached. If you want to check null, it should be first to make sure it is before the comparison